agent/taskresource/asmauth/asmauthstatus_test.go

```diff
@@ -0,0 +1,90 @@
+// +build unit
+
+// Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package asmauth
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStatusString(t *testing.T) {
+	var resourceStatus ASMAuthStatus
+
+	resourceStatus = ASMAuthStatusNone
+	assert.Equal(t, resourceStatus.String(), "NONE")
+	resourceStatus = ASMAuthStatusCreated
+	assert.Equal(t, resourceStatus.String(), "CREATED")
+	resourceStatus = ASMAuthStatusRemoved
+	assert.Equal(t, resourceStatus.String(), "REMOVED")
+}
+
+func TestMarshalASMAuthStatus(t *testing.T) {
+	status := ASMAuthStatusNone
```

Comment:
Can you also test "ASMAuthStatusCreated"? And this can be combined together with below using table test.